lib/index.js

```diff
@@ -0,0 +1,233 @@
+   * Generates a KeyPair with an optional deterministic seed.
+   * @param {KeyPairOptions} [options={}] - The options to use.
+   *
+   * @returns {Promise<Secp256KeyPair>} Generates a key pair.
+   */
+  static async generate(options = {}) {
+
+    const key = ec.genKeyPair();
+    const pubPoint = key.getPublic();
+    // encode public X and Y in compressed form
+    const publicKeyBase58 = base58.encode(new Uint8Array(
+      pubPoint.encodeCompressed()));
+    const privateKeyBase58 = base58.encode(new Uint8Array(
+      key.getPrivate().toArray()));
+    return new Secp256KeyPair({
+      privateKeyBase58,
+      publicKeyBase58,
+      ...options
+    });
+  }
+
+  /**
+   * Returns a signer object for use with jsonld-signatures.
+   *
+   * @returns {{sign: Function}} A signer for the json-ld block.
+   */
+  signer() {
+    return secp256SignerFactory(this);
+  }
+
+  /**
+   * Returns a verifier object for use with jsonld-signatures.
+   *
+   * @returns {{verify: Function}} Used to verify jsonld-signatures.
+   */
+  verifier() {
+    return secp256VerifierFactory(this);
+  }
+
+  /**
+   * Adds a public key base to a public key node.
+   *
+   * @param {Object} publicKeyNode - The public key node in a jsonld-signature.
+   * @param {string} publicKeyNode.publicKeyBase58 - Base58 Public Key for
+   * jsonld-signatures.
+   *
+   * @returns {{verify: Function}} A PublicKeyNode in a block.
+   */
+  addEncodedPublicKey(publicKeyNode) {
+    publicKeyNode.publicKeyBase58 = this.publicKeyBase58;
+    return publicKeyNode;
+  }
+
+  /**
+   * Generates and returns a public key fingerprint.
+   *
+   * @param {string} publicKeyBase58 - The base58 encoded public key material.
+   *
+   * @returns {string} The fingerprint.
+   */
+  static fingerprintFromPublicKey(/*{publicKeyBase58}*/) {
+    // TODO: implement
+    throw new Error('`fingerprintFromPublicKey` API is not implemented.');
+  }
+
+  /**
+   * Generates and returns a public key fingerprint.
+   *
+   * @returns {string} The fingerprint.
+   */
+  fingerprint() {
+    // TODO: implement
+    throw new Error('`fingerprint` API is not implemented.');
+  }
+
+  /**
+   * Tests whether the fingerprint was generated from a given key pair.
+   *
+   * @param {string} fingerprint - A Base58 public key.
+   *
+   * @returns {Object} An object indicating valid is true or false.
+   */
+  verifyFingerprint(/*fingerprint*/) {
+    // TODO: implement
+    throw new Error('`verifyFingerprint` API is not implemented.');
+  }
+
+  static async from(options) {
+    return new Secp256KeyPair(options);
+  }
+
+  /**
+   * Contains the public key for the KeyPair
+   * and other information that json-ld Signatures can use to form a proof.
+   * @param {Object} [options={}] - Needs either a controller or owner.
+   * @param {string} [options.controller=this.controller]  - DID of the
+   * person/entity controlling this key pair.
+   * @param {string} [options.owner=this.owner] - DID of key owner.
```

Comment:
Let's not include `owner` at all ... it's old stuff and this is a new key pair that shouldn't need the backwards compat.